src/MongoDB.Driver/ClusterKey.cs

```diff
@@ -171,13 +180,15 @@ public override bool Equals(object obj)
                 _heartbeatInterval == rhs._heartbeatInterval &&
                 _heartbeatTimeout == rhs._heartbeatTimeout &&
                 _ipv6 == rhs._ipv6 &&
+                GetValueOrEmpty(_kmsProviders).SequenceEqual(GetValueOrEmpty(rhs._kmsProviders), new KmsProvidersComparer()) &&
                 _localThreshold == rhs._localThreshold &&
                 _maxConnectionIdleTime == rhs._maxConnectionIdleTime &&
                 _maxConnectionLifeTime == rhs._maxConnectionLifeTime &&
                 _maxConnectionPoolSize == rhs._maxConnectionPoolSize &&
                 _minConnectionPoolSize == rhs._minConnectionPoolSize &&
                 _receiveBufferSize == rhs._receiveBufferSize &&
                 _replicaSetName == rhs._replicaSetName &&
+                GetValueOrEmpty(_schemaMap).SequenceEqual(GetValueOrEmpty(rhs._schemaMap)) &&
```

Comment:
What is SequenceEqual comparing? I assume KeyValueParis.

Does KeyValuePair implement Equals properly?